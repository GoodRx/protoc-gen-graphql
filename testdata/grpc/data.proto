syntax = "proto3";

package protoc_gen_graphql.test.grpc;

message RequestB {
  string id = 1;
}

message ResponseB {
  string id = 1;
}
