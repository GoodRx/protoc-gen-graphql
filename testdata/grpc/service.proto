syntax = "proto3";

package protoc_gen_graphql.test.grpc;

import "grpc/data.proto";

service ServiceA {
  rpc MethodA(RequestA) returns (ResponseA);
  rpc MethodB(RequestB) returns (ResponseB);
}

message RequestA {
  message RequestAA {
    message RequestAAA {
      string aaa = 1;
    }
  }

  string string_field = 1;
  RequestAA request_aa = 2;
  repeated RequestAA.RequestAAA request_aaa = 3;
  map<string, EnumType> map_field = 4;
  oneof oneof_field {
    string first_oneof_value = 5;
    RequestAA second_oneof_value = 6;
  }
}

message ResponseA {
  string string_field = 1;
}

enum EnumType {
  UNKNOWN = 0;
  FIRST = 1;
  SECOND = 2;
}
