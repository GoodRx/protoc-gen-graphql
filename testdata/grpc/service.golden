# DO NOT EDIT! Generated by protoc-gen-graphql.

type ProtocGenGraphqlTestGrpc_ServiceA_Query {
  MethodA(input: ProtocGenGraphqlTestGrpc_RequestA_Input!): ProtocGenGraphqlTestGrpc_ResponseA!
  MethodB(input: ProtocGenGraphqlTestGrpc_RequestB_Input!): ProtocGenGraphqlTestGrpc_ResponseB!
}

type ProtocGenGraphqlTestGrpc_RequestA {
  string_field: String!
  request_aa: Boolean
  request_aaa: [ProtocGenGraphqlTestGrpc_RequestA_RequestAA_RequestAAA]!
  map_field: [ProtocGenGraphqlTestGrpc_RequestA_MapField!]!
  oneof_field: ProtocGenGraphqlTestGrpc_RequestA_OneofField
}

union ProtocGenGraphqlTestGrpc_RequestA_OneofField = ProtocGenGraphqlTestGrpc_RequestA_OneofField_FirstOneofValue | ProtocGenGraphqlTestGrpc_RequestA_OneofField_SecondOneofValue

type ProtocGenGraphqlTestGrpc_RequestA_OneofField_FirstOneofValue {
  first_oneof_value: String!
}

type ProtocGenGraphqlTestGrpc_RequestA_OneofField_SecondOneofValue {
  second_oneof_value: Boolean
}

input ProtocGenGraphqlTestGrpc_RequestA_Input {
  string_field: String!
  request_aa: Boolean
  request_aaa: [ProtocGenGraphqlTestGrpc_RequestA_RequestAA_RequestAAA_Input]!
  map_field: [ProtocGenGraphqlTestGrpc_RequestA_MapField_Input!]!
  oneof_field: ProtocGenGraphqlTestGrpc_RequestA_OneofField_Input
}

input ProtocGenGraphqlTestGrpc_RequestA_OneofField_Input {
  first_oneof_value: String
  second_oneof_value: Boolean
}

type ProtocGenGraphqlTestGrpc_RequestA_RequestAA_RequestAAA {
  aaa: String!
}

input ProtocGenGraphqlTestGrpc_RequestA_RequestAA_RequestAAA_Input {
  aaa: String!
}

type ProtocGenGraphqlTestGrpc_RequestA_MapField {
  key: String!
  value: ProtocGenGraphqlTestGrpc_EnumType!
}

input ProtocGenGraphqlTestGrpc_RequestA_MapField_Input {
  key: String!
  value: ProtocGenGraphqlTestGrpc_EnumType!
}

type ProtocGenGraphqlTestGrpc_ResponseA {
  string_field: String!
}

enum ProtocGenGraphqlTestGrpc_EnumType {
  UNKNOWN
  FIRST
  SECOND
}
